src/book/mod.rs

```diff
@@ -149,14 +154,23 @@ impl MDBook {
     pub fn build(&self) -> Result<()> {
         debug!("[fn]: build");
 
+        let mut preprocessed_book = self.book.clone();
```

Comment:
I like that we're not mutating the original book here and instead preprocessing its clone :+1: